### ISSUE TITLE
Double free or corruption in the exit

### ISSUE THREAD

**Issue by sergiobenrocha2** [mgba-emu/mgba#206]
I'm getting an error in the Qt exit on ARM. Missing some symbols yet.

```
Starting program: /usr/games/mgba-qt 
Traceback (most recent call last):
  File "/usr/share/gdb/auto-load/usr/lib/arm-linux-gnueabihf/libstdc++.so.6.0.19-gdb.py", line 63, in <module>
    from libstdcxx.v6.printers import register_libstdcxx_printers
ImportError: No module named 'libstdcxx'
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
[New Thread 0xb24da410 (LWP 14845)]
[New Thread 0xb1cda410 (LWP 14846)]
[New Thread 0xb066f410 (LWP 14848)]
[New Thread 0xafcff410 (LWP 14849)]
[New Thread 0xaf4ff410 (LWP 14850)]
[New Thread 0xae8ff410 (LWP 14851)]
[Thread 0xae8ff410 (LWP 14851) exited]
libGL error: MESA-LOADER: malformed or no PCI ID
libGL error: dlopen /usr/lib/arm-linux-gnueabihf/dri/exynos_dri.so failed (/usr/lib/arm-linux-gnueabihf/dri/exynos_dri.so: cannot open shared object file: No such file or directory)
libGL error: dlopen ${ORIGIN}/dri/exynos_dri.so failed (${ORIGIN}/dri/exynos_dri.so: cannot open shared object file: No such file or directory)
libGL error: dlopen /usr/lib/dri/exynos_dri.so failed (/usr/lib/dri/exynos_dri.so: cannot open shared object file: No such file or directory)
libGL error: unable to load driver: exynos_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: exynos
[New Thread 0xae8ff410 (LWP 14852)]
[New Thread 0xad273410 (LWP 14853)]
[New Thread 0xa7979410 (LWP 14854)]
[Thread 0xa7979410 (LWP 14854) exited]
[New Thread 0xa7979410 (LWP 14855)]
[Thread 0xad273410 (LWP 14853) exited]
[Thread 0xae8ff410 (LWP 14852) exited]
*** Error in `/usr/games/mgba-qt': double free or corruption (!prev): 0x003b0920 ***

Program received signal SIGABRT, Aborted.
__libc_do_syscall () at ../ports/sysdeps/unix/sysv/linux/arm/libc-do-syscall.S:44
44	../ports/sysdeps/unix/sysv/linux/arm/libc-do-syscall.S: No such file or directory.
(gdb) 
(gdb) bt full
#0  __libc_do_syscall () at ../ports/sysdeps/unix/sysv/linux/arm/libc-do-syscall.S:44
No locals.
#1  0xb614ef0e in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
        _a1 = 0
        _a3tmp = 6
        _a1tmp = 0
        _a3 = 6
        _nametmp = 268
        _a2tmp = 14842
        _a2 = 14842
        _name = 268
        _sys_result = <optimized out>
        pd = 0xb288a000
        pid = 0
        selftid = 14842
#2  0xb6151766 in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, sa_mask = {__val = {3039308316, 1, 5, 3040487952, 3040487512, 3039310580, 3204441832, 
              996378008, 3204441980, 3204441864, 3204441872, 3070124733, 3204441872, 3070229172, 0, 0, 5, 0, 1, 3040487512, 3204441904, 3070229172, 0, 0, 0, 
              3040487952, 1, 3040487512, 4294967295, 0, 3039308316, 3040487512}}, sa_flags = 0, sa_restorer = 0xbeffe818}
        sigs = {__val = {32, 0 <repeats 31 times>}}
#3  0xb6175474 in __libc_message (do_abort=<optimized out>, fmt=0xb61f6904 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
        ap = {__ap = 0xbeffe864}
        fd = 23
        on_2 = <optimized out>
        list = <optimized out>
        nlist = <optimized out>
        cp = <optimized out>
        written = <optimized out>
#4  0xb617c022 in malloc_printerr (action=1, str=0xb61f6a38 "double free or corruption (!prev)", ptr=<optimized out>) at malloc.c:4996
        buf = "003b0920"
        cp = <optimized out>
#5  0xb617ca48 in _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3840
        size = <optimized out>
        fb = <optimized out>
        nextchunk = <optimized out>
        nextsize = <optimized out>
        nextinuse = <optimized out>
        prevsize = <optimized out>
        bck = <optimized out>
        fwd = <optimized out>
        errstr = <optimized out>
        locked = <optimized out>
        __func__ = "_int_free"
#6  0x0004f714 in QGBA::GameController::~GameController (this=0x35b780, __in_chrg=<optimized out>)
    at /home/odroid/Projects/mGBA/mgba/src/platform/qt/GameController.cpp:222
No locals.
#7  0x0004f814 in QGBA::GameController::~GameController (this=0x35b780, __in_chrg=<optimized out>)
    at /home/odroid/Projects/mGBA/mgba/src/platform/qt/GameController.cpp:225
No locals.
#8  0xb64f5da0 in QObjectPrivate::deleteChildren() () from /usr/lib/arm-linux-gnueabihf/libQt5Core.so.5
No symbol table info available.
#9  0xb6b11490 in QWidget::~QWidget() () from /usr/lib/arm-linux-gnueabihf/libQt5Widgets.so.5
No symbol table info available.
#10 0xb6bc8bbe in QMainWindow::~QMainWindow() () from /usr/lib/arm-linux-gnueabihf/libQt5Widgets.so.5
No symbol table info available.
#11 0x0008c9fe in QGBA::Window::~Window (this=0x2daea0, __in_chrg=<optimized out>) at /home/odroid/Projects/mGBA/mgba/src/platform/qt/Window.cpp:173
No locals.
#12 0x0008ca3c in QGBA::Window::~Window (this=0x2daea0, __in_chrg=<optimized out>) at /home/odroid/Projects/mGBA/mgba/src/platform/qt/Window.cpp:173
No locals.
#13 0xb64f61f8 in QObject::event(QEvent*) () from /usr/lib/arm-linux-gnueabihf/libQt5Core.so.5
No symbol table info available.
#14 0xb6b135c6 in QWidget::event(QEvent*) () from /usr/lib/arm-linux-gnueabihf/libQt5Widgets.so.5
No symbol table info available.
#15 0xb6bc9e22 in QMainWindow::event(QEvent*) () from /usr/lib/arm-linux-gnueabihf/libQt5Widgets.so.5
No symbol table info available.
#16 0xb6aeb8a0 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/arm-linux-gnueabihf/libQt5Widgets.so.5
No symbol table info available.
#17 0xb6aef36e in QApplication::notify(QObject*, QEvent*) () from /usr/lib/arm-linux-gnueabihf/libQt5Widgets.so.5
No symbol table info available.
#18 0xbeffee28 in ?? ()
No symbol table info available.
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```

**Comment by endrift** [comment]
Should be fixed in 3fef28db198e8d1397a9aed3238b1ebf8c7352b1. Let me know.

**Comment by sergiobenrocha2** [comment]
It's working, thanks.
